### PR TITLE
Retain cycle rule

### PR DESCRIPTION
**Pull request by ThibaultKlein**

### CODE REVIEW

**Comment by htinlinn** [comment]
Inviting myself to your PR here. But, doesn't this fit more in Best Practices/Tips and Tricks section than in the main style guide?

**Comment by Haud** [comment]
@htinlinn invite yourself to every PR :)

File: README.md [hunk 1]
```diff
@@ -418,6 +418,44 @@ func isError(error: Error?) -> Bool {
...
+```
+internal final class myClass {
+    weak var delegate: MyDelegate?
+
+    func myFunction() {
+        delegate?.doSomething()
+    }
+    
+    func myFunctionWithClosure() {
+    	let closure = { [weak self, unowned myClassInstance] in
```
Comment (Haud) [inline README.md]: what is `myClassInstance`

Comment (ThibaultKlein) [inline README.md]: just a dummy class instance to demonstrate that here we want to use unowned since this class will not be nil

**Comment by ThibaultKlein** [comment]
@htinlinn I think thinking about retain cycle should be a standard at Prolific, that's something we often forget to do, we should be very careful about it. IMHO it should live in the Swift rules.

**Comment by htinlinn** [comment]
@ThibaultKlein totally agree. We should probably even check for them as part of the audits that we do. But to me, this feels like something you should be doing anyways vs a nicer way of doing things which is what most of the style guide has been about.

**Comment by Haud** [comment]
Agree with Linn on this. There is valid information here, but this isn't an enforcable style as much as it is an aspect others might not be aware of.

File: README.md (hunk 1 above)
Comment (paulmiard) [inline README.md]: maybe add an inline comment to make this clear?

**Comment by ThibaultKlein** [comment]
@Haud @htinlinn I understand your points, I updated to move it to best practices

**Comment by Haud** [comment]
:+1: 

Looks good to me. HOw about you @htinlinn ?

**Comment by paulmiard** [comment]
Next thing we add we should add a toc to best practices

**Comment by paulmiard** [comment]
:+1:

File: BestPractices.md [hunk 2]
```diff
@@ -53,3 +53,42 @@ class myViewController: UIViewController {
 let myViewController = UIViewController()
 myViewController.text = "Prolific" // Crash because the view controller label has not been initialized yet
 ```
+
+### Retain cycle ###
```
Comment (htinlinn) [inline BestPractices.md]: Make this title case?

File: BestPractices.md [hunk 3]
```diff
@@ -53,3 +53,42 @@ class myViewController: UIViewController {
...
+    weak var delegate: MyDelegate?
+
+    func myFunction() {
+        delegate?.doSomething()
+    }
+    
+    func myFunctionWithClosure() {
+    	let closure = { [weak self, unowned notNilInstance] in
+	    self?.doSomething() // weak variables are optionals
+    	notNilInstance.doSomethingElse() //unowned variables are not
```
Comment (htinlinn) [inline BestPractices.md]: Put a space after //?

**Comment by Haud** [comment]
@ThibaultKlein @htinlinn has this been updated based on Linn's comments? If not, can we get that sorted as soon as possible, por favor?

**Comment by ThibaultKlein** [comment]
Completely forgot about this guy, I updated

File: BestPractices.md (hunk 2 above)
Comment (htinlinn) [inline BestPractices.md]: @ThibaultKlein i think you missed this one.

Comment (ThibaultKlein) [inline BestPractices.md]: my bad, fixed

**Comment by ThibaultKlein** [comment]
@Haud @htinlinn ?

**Comment by Haud** [comment]
Gave my :+1: before; @htinlinn is the last hold out.

**Comment by htinlinn** [comment]
:+1:

**Pull request merged**